kubeflow/core/cert-manager.libsonnet

```diff
@@ -0,0 +1,181 @@
+      },
+      spec: {
+        group: "certmanager.k8s.io",
+        version: "v1alpha1",
+        names: {
+          kind: "Issuer",
+          plural: "issuers",
+        },
+        scope: "Namespaced",
+      },
+    },
+    
+    serviceAccount:: {
+      apiVersion: "v1",
+      kind: "ServiceAccount",
+      metadata: {
+        name: "cert-manager",
+        namespace: namespace,
+      }
+    },
+
+    clusterRole:: {
+      apiVersion: "rbac.authorization.k8s.io/v1beta1",
+      kind: "ClusterRole",
+      metadata: {
+        name: "cert-manager",
+      },
+      rules: [
+        {
+          apiGroups: ["certmanager.k8s.io"],
+          resources: ["certificates", "issuers", "clusterissuers"],
+          verbs: ["*"],
+        },
+        {
+          apiGroups: [""],
+          resources: ["secrets", "events", "endpoints", "services", "pods"],
+          verbs: ["*"],
+        },
+        {
+          apiGroups: ["extensions"],
+          resources: ["ingresses"],
+          verbs: ["*"],
+        },
+      ],
+    },
+
+    clusterRoleBinding:: {
+      apiVersion: "rbac.authorization.k8s.io/v1beta1",
+      kind: "ClusterRoleBinding",
+      metadata: {
+        name: "cert-manager",
+      },
+      roleRef: {
+        apiGroup: "rbac.authorization.k8s.io",
+        kind: "ClusterRole",
+        name: "cert-manager",
+      },
+      subjects: [
+        {
+          name: "cert-manager",
+          namespace: namespace,
+          kind: "ServiceAccount",
+        },
+      ],
+    },
+
+    deploy:: {
+      apiVersion: "apps/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        name: "cert-manager",
+        namespace: namespace,
+        labels: {
+          app: "cert-manager",
+        },
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "cert-manager",
+            },
+          },
+          spec: {
+            serviceAccountName: "cert-manager",
+            containers: [
+              {
+                name: "cert-manager",
+                image: "quay.io/jetstack/cert-manager-controller:v0.2.3",
+                imagePullPolicy: "IfNotPresent",
+              },
+              {
+                name: "ingress-shim",
+                image: "quay.io/jetstack/cert-manager-ingress-shim:v0.2.3",
+                imagePullPolicy: "IfNotPresent",
+              },
+            ],
+          },
+        },
+      },
+    },
+
+    issuerLEProd(acmeEmail, acmeUrl):: {
+      apiVersion: "certmanager.k8s.io/v1alpha1",
+      kind: "Issuer",
+      metadata: {
+        name: "letsencrypt-prod",
+        namespace: namespace,
+      },
+      spec: {
+        acme: {
+          server: acmeUrl,
+          email: acmeEmail,
+          privateKeySecretRef: {
+            name: "letsencrypt-prod-secret",
+          },
+          http01: {
+            // Dummy field used to jsonnet does not remove empty 'http01: {}' entry.
```

Comment:
You can get rid of `std.prune` at this top of this file and add a comment explaining why

Then you don't need a dummy field.

Reply:
I was wondering how to get around that. :) thank you!